src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.WithFrozenSourceGeneratedDocumentsCompilationTracker.cs

```diff
@@ -46,7 +46,7 @@ private sealed class WithFrozenSourceGeneratedDocumentsCompilationTracker : ICom
         private SkeletonReferenceCache _skeletonReferenceCache;
 
         public WithFrozenSourceGeneratedDocumentsCompilationTracker(
-            ICompilationTracker underlyingTracker,
+            RegularCompilationTracker underlyingTracker,
```

Comment:
primary goal was this change.  ensuring that we always are only wrapping a single RegularCompilationTracker, and we are *not* wrapping a WithFrozenSourceGeneratedDocumentsCompilationTracker with another WithFrozenSourceGeneratedDocumentsCompilationTracker.